File: Python/49GroupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = []
        maps = {}
        for word in strs:
            charArray = [0] * 26
            for c in word:
                # translate char to index and inc count
                charArray[ord(c) - 97] += 1
            if tuple(charArray) in maps:
                maps[tuple(charArray)].append(word)
            else:
                maps[tuple(charArray)] = [word]
        
        for value in maps.values():
            res.append(value)
        return res

File: Swift/49GroupAnagrams.swift
class Solution {
    func groupAnagrams(_ strs: [String]) -> [[String]] {
        
        // dictionary with key as sorted string and values as original
        var dict = [String : [String]]()
        for word in strs {
            let key = String(word.sorted())
            if dict[key] != nil {
                // we have an anagram, add word to list
                dict[key, default: []].append(word)
            } else {
                // else set the key to array of word
                dict[key] = [word]
            }
        }
        return Array(dict.values)
    }
}
